File: 0496.下一个更大元素I/0496-下一个更大元素I.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        #ÎÒµÄ´úÂë
        for i in range(len(nums1)):
            for j in range(len(nums2)):
                if nums1[i]==nums2[j]:
                    if len(nums2)-1-j==0:
                        temp=-1
                    else:
                        for t in range(j+1,len(nums2)):
                            if nums2[t]>nums1[i]:
                                temp=nums2[t]
                                # nums1[i]=temp
                                break
                            else:
                                temp=-1
            nums1[i]=temp             
        return nums1

File: 0496.下一个更大元素I/0496-下一个更大元素I.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int> umap;
        for( int i = 0; i < nums1.size(); i++){
            umap[nums1[i]] = i;
        }
        vector<int> answer(nums1.size(),-1);
        stack<int> st;
        st.push(0);
        for( int i = 1; i < nums2.size(); i++){
            if( nums2[i] <= nums2[st.top()]){
                st.push(i);
            }else{
                while( !st.empty() && nums2[i] > nums2[st.top()]){
                    auto iter = umap.find(nums2[st.top()]);
                    if( iter != umap.end()){
                        answer[iter->second] = nums2[i];
                    }
                    st.pop();
                }
                st.push(i);
            }
        }
        return answer;

    }
};